azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -571,7 +614,57 @@ def init_poolmanager(self, *args, **kwargs):
         super(HTTPAdapterWithSocketOptions, self).init_poolmanager(*args, **kwargs)
 
 
-class KustoClient:
+class _KustoClientBase:
+    _mgmt_default_timeout = timedelta(hours=1, seconds=30)
+    _query_default_timeout = timedelta(minutes=4, seconds=30)
+    _streaming_ingest_default_timeout = timedelta(minutes=10)
+
+    def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str]):
+        self._kcsb = kcsb
+        if not isinstance(kcsb, KustoConnectionStringBuilder):
+            self._kcsb = KustoConnectionStringBuilder(kcsb)
+        self._kusto_cluster = self._kcsb.data_source
+
+        # Create a session object for connection pooling
+        self._mgmt_endpoint = "{0}/v1/rest/mgmt".format(self._kusto_cluster)
+        self._query_endpoint = "{0}/v2/rest/query".format(self._kusto_cluster)
+        self._streaming_ingest_endpoint = "{0}/v1/rest/ingest/".format(self._kusto_cluster)
+        self._request_headers = {"Accept": "application/json", "Accept-Encoding": "gzip,deflate", "x-ms-client-version": "Kusto.Python.Client:" + VERSION}
+
+    @staticmethod
+    def _kusto_parse_by_endpoint(endpoint: str, response_json: Any) -> KustoResponseDataSet:
+        if endpoint.endswith("v2/rest/query"):
+            return KustoResponseDataSetV2(response_json)
+        return KustoResponseDataSetV1(response_json)
+
+    @staticmethod
+    def _handle_http_error(
+        exception: "Union[HTTPError, aiohttp.web_exceptions.HTTPError]",
+        endpoint: Optional[str],
+        payload: Optional[io.IOBase],
+        response: "Union[Response, aiohttp.ClientResponse]",
+        response_json: Any,
+        response_text: Optional[str],
+    ) -> NoReturn:
+        if response.status_code == 404:
+            if payload:
+                raise KustoServiceError("The ingestion endpoint does not exist. Please enable streaming ingestion on your cluster.", response) from exception
+            else:
+                raise KustoServiceError("The requested endpoint '{}' does not exist.".format(endpoint), response) from exception
+        if payload:
+            raise KustoServiceError(
+                "An error occurred while trying to ingest: Status: {0.status_code}, Reason: {0.reason}, Text: {1}.".format(response, response_text), response
+            ) from exception
+        try:
+            raise KustoServiceError([response_json], response) from exception

```

Comment:
That is super weird.. let talk offline..

Reply:
I think it's good now